doc: update documentation and definitions files (#154)
dashevo/wallet-lib: docs/getting-started/core-concepts.md

Since the introduction of [deterministic wallet](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki), a Wallet is actually composed of multiple account. 

For manipulating multiple accounts, `Wallet.getAccount()` takes optional [options](/wallet/getAccount.md) where index allows either accessing or creating a specific account index. 

## Instantiation types

A Wallet instance can be created from multiples types, which impact how much the Wallet can do. 
In general, we expect you to initialize from a `mnemonic` or an `seed` (HD seed) or an `HDPrivateKey`, which allows wallet-lib to deal with HD Wallet (deterministic wallet).  

In some other cases, you might want to instantiate Wallet from another input such as : 

> - `privateKey`: This allows to manage a single privateKey/publicKey set. Therefore, you will only have a single unique address to receive money. 

Reviewer: ```suggestion
- `privateKey`: This allows managing a single privateKey/publicKey set. Therefore, you will only have a single unique address to receive money. 
```

> - `HDPublicKey`: This allows a "watch-only" mode. You won't be able to spend anything, but this will allow you to track and monitor in real-time the addresses set of this public key. Use-case might be to always pay to another new 

Reviewer: Looks like the second sentence was truncated? "...pay to another new..." address? wallet?
```suggestion
- `HDPublicKey`: This allows a "watch-only" mode. You won't be able to spend anything, but this will allow you to track and monitor in real-time the address set of this public key. Use-case might be to always pay to another new 
```

Reviewer: @Alex-Werner I combined a comment and a suggestion so you probably missed the comment. It looks like there should be more to the second sentence ("Use-case might be to always pay to another new")?

Since the introduction of [deterministic wallet](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki), a Wallet is actually composed of multiple account. 

> For manipulating multiple accounts, `Wallet.getAccount()` takes optional [options](/wallet/getAccount.md) where index allow to either access or create a specific account index. 

Reviewer: ```suggestion
For manipulating multiple accounts, `Wallet.getAccount()` takes optional [options](/wallet/getAccount.md) where index allows either accessing or creating a specific account index. 
```

In some other cases, you might want to instantiate Wallet from another input such as : 
- `privateKey`: This allows managing a single privateKey/publicKey set. Therefore, you will only have a single unique address to receive money. 

> - `HDPublicKey`: This allows a "watch-only" mode. You won't be able to spend anything, but this will allow you to track and monitor in real-time the address set of this public key. This allows you to derivate, from a single key (shared by another user, third-party merchant), a unique unused addresses.  

Reviewer: ```suggestion
- `HDPublicKey`: This allows a "watch-only" mode. You won't be able to spend anything, but this will allow you to track and monitor in real-time the address set of this public key. This allows you to derive unique unused addresses from a single key (shared by another user, third-party merchant).  
```

- `privateKey`: This allow to manage a single privateKey/publicKey set. Therefore, you will only have a single unique address to receive money. 
- `HDPublicKey`: This allow a "watch-only" mode. You won't be able to spend anything, but this will allow you to track and monitor in real-time the addresses set of this public key. Use-case might be to always pay to another new 

> [Learn more about instantiation documentation](/wallet/Wallet.md)

Reviewer: Is `/wallet/Wallet.md` correct? It doesn't work if you click it when viewing the md, but maybe it does when docsify builds the docs?

Author: Yes, this is an issue we had with a PR suggested by pasta in the past. 
We can make a direct link to the full path, but if we go this route, it requires to be done for all the similar link. 

What do you think @thephez ? I do find some good reason to provide it for when we actually browse from github (not on github.io), in the other end, if one day we get rid of github.io, it will not work, and it makes it less clear for when we browse on an offline condition 🤔

Reviewer: I think it's fine to leave it the way that works on the github.io page. That's the preferred place to look at it. 

Also, it's easy enough to view docsify docs locally. Just run:
`cd docs && python -m SimpleHTTPServer 3000` 
from the repo root as described in the [docsify quickstart](https://docsify.js.org/#/quickstart?id=manual-initialization). Another offline option would be to look at the [PWA config](https://docsify.js.org/#/pwa).
